rust/Cargo.lock

```diff
@@ -0,0 +1,799 @@
+# This file is automatically @generated by Cargo.
```

Comment:
I'm not totally convinced that `Cargo.lock` must be checked-in for libraries.

After checking some of the most popular crates, almost none of them have it (arrow-rs neither). 

On the contrary, I can't find any good reason to not have it 😃

Reply:
More info about this: 
- https://blog.rust-lang.org/2023/08/29/committing-lockfiles.html
- https://doc.rust-lang.org/nightly/cargo/faq.html#why-have-cargolock-in-version-control.